content/en/docs/concepts/workloads/controllers/job.md

```diff
@@ -659,6 +674,51 @@ there is no automatic Job restart once the Job status is `type: Failed`.
 That is, the Job termination mechanisms activated with `.spec.activeDeadlineSeconds`
 and `.spec.backoffLimit` result in a permanent Job failure that requires manual intervention to resolve.
 
+### Terminal Job conditions
+
+A Job has two possible terminal states, it ends up either succeeded, or failed,
+and these states are reflected by the presence of the Job conditions `Complete`
+or `Failed`, respectively.
+
+The failure scenarios encompass:
+- the number of pod failures exceeded the specified `.spec.backoffLimit`;
+  see also [Pod backoff failure policy](#pod-backoff-failure-policy)
+- the Job runtime exceeded the specified `.spec.activeDeadlineSeconds`
+- the Job using `.spec.backoffLimitPerIndex` has some indexes failed;
+  see also [Backoff limit per index](#backoff-limit-per-index)
+- the number of failed indexes in the Job has exceeded the specified
+  `spec.maxFailedIndexes`; see also [Backoff limit per index](#backoff-limit-per-index)
+- a failed Pod matches a rule in `.spec.podFailurePolicy`
+   with the `FailJob` action; see also [Pod failure policy](#pod-failure-policy) for more details on how the policy rules can affect evaluation of failures
+
+The success scenarios encompass:
+- the number of succeeded pods reached the specified `.spec.completions`
+- the criteria specified by the `.spec.successPolicy` are met;
+  see also [Success policy](#success-policy)
+
+### Termination of Job pods
+
+Prior to v1.31 the Job terminal conditions are added when the Job termination
+process is triggered, and all Pod finalizers are removed, but some pods may
+still remain running/terminating at that point in time.
```

Comment:
I think it would be useful to keep trace of the change in one place, as this is a substantial logic change, and even though the docs are versioned users may not have a clue something has changed. We did it in the past for numerous features, for example: https://kubernetes.io/docs/concepts/cluster-administration/node-shutdown/.

That being said, I make sure the version is mentioned in only one place (before review it was scattered across).